src/axom/sidre/CMakeLists.txt

```diff
@@ -91,37 +91,29 @@ if (ENABLE_FORTRAN)
     endif()
 endif()
 
-################################
-# Make/Install the library
-################################
+#------------------------------------------------------------------------------
+# Build and install the library
+#------------------------------------------------------------------------------
 set(sidre_depends
     core
     conduit::conduit
     slic)
 
-if(HDF5_FOUND)
-    list(APPEND sidre_depends hdf5)
-endif()
+blt_list_append(TO sidre_depends ELEMENTS hdf5 IF HDF5_FOUND)
+blt_list_append(TO sidre_depends ELEMENTS sparsehash IF SPARSEHASH_FOUND)
 
 if(ENABLE_MPI)
     list(APPEND sidre_depends fmt mpi)
```

Comment:
It's strange that fmt is guarded by ENABLE_MPI...

Reply:
Yes, that is startling.

Reply:
Oof, that's probably my fault.  It should be in the default sidre_depends definition a few lines above.

Reply:
I understood that to mean that ``fmt`` was a dependency of ``spio`` but not of ``sidre``.

It is probably best to just add a comment above ``if(ENABLE_MPI)`` about the extra dependencies for ``spio``, rather than adding additional unnecessary dependencies to ``sidre``.

Reply:
I added some comments around the ``if(ENABLE_MPI)`` guards.